hkmc2/shared/src/main/scala/hkmc2/bbml/bbML.scala

```diff
@@ -478,7 +451,7 @@ class BBTyper(using elState: Elaborator.State, tl: TL):
   // * Try to instantiate the given type if it is forall quantified
   private def tryMkMono(ty: GeneralType, sc: Located)(using Ctx): Type = ty match
     case pt: PolyType => tryMkMono(instantiate(pt), sc)
-    case ft: PolyFunType => ft.monoOr(error(msg"Can not instantiate ${ty.toString} to mono!." -> sc.toLoc :: Nil))
+    case ft: PolyFunType => ft.monoOr(error(msg"Can not instantiate a polymorphic type inside ${ty.toString}" -> sc.toLoc :: Nil))
```

Comment:
Error message is still terrible. Think about some random user who knows nothing seeing this sort of things. By the way, it's "cannot", not "can not".